clientpool/channel.go

```diff
@@ -11,23 +10,17 @@ import (
 )
 
 type channelPool struct {
-	pool           chan Client
-	opener         ClientOpener
-	numActive      atomic.Int32
-	initialClients int
-	maxClients     int
+	pool       chan Client
+	opener     ClientOpener
+	numActive  atomic.Int32
+	maxClients int
 }
 
 // Make sure channelPool implements Pool interface.
 var _ Pool = (*channelPool)(nil)
 
 // NewChannelPool creates a new client pool implemented via channel.
-//
-// Note that this function could return both non-nil Pool and error,
-// when we failed to create all asked bestEffortInitialClients.
-// In such case the returned Pool would have at least requiredInitialClients
-// clients we already established.
-func NewChannelPool(ctx context.Context, requiredInitialClients, bestEffortInitialClients, maxClients int, opener ClientOpener) (Pool, error) {
+func NewChannelPool(ctx context.Context, requiredInitialClients, bestEffortInitialClients, maxClients int, opener ClientOpener) (_ Pool, err error) {
```

Comment:
Interesting, never thought wildcard can be used to have partially named result parameters.